### DISCUSSION TITLE
Different Scenario for Human Workflow with Elsa  Core

### DISCUSSION THREAD

**Discussion by r-ulak** [elsa-workflows/elsa-core#3258] [Q&A]
I am planning to migrate my current custom built Human workflows to instead use Elsa. I am looking to do this via code behdind . 
I have scanned thru lot of sample projects in this repo and could not find examples on how to

1. Send the Workflow back to previous steps (Send Back to initiator / for review/ correction)
2. Reassign the Step to another Owner ( current User its waiting on could be Out of office )
3. Hierarchical approval (it goes to multi level approval until it reaches CEO or a Dept head etc) -- (this i can see how it can be built).

Can these task be done in Elsa 2.8  backend in c# without using designer?

**Comment by Nokecy** [comment]
@sfmskywalker  in elsa v3  Can it be achieved

**Comment by sfmskywalker** [comment]
Hi @r-ulak and @Nokecy,

Certainly, Elsa 2 and 3 offer the flexibility to build workflows that encompass human interactions. To achieve this, you'll need to craft custom activities and possibly custom controllers to seamlessly interact with the Elsa workflow engine.

While I might not have a comprehensive example at hand, the following resources should provide a foundational understanding of how to bridge interactions between the Elsa workflow engine and your application:

- [Elsa Document Approval Guide](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval)
- [Building Workflow-Driven .NET Applications with Elsa 2 - Part 1](https://sipkeschoorstra.medium.com/building-workflow-driven-net-applications-with-elsa-2-part-1-44e08a9ba94b)

To address the specific tasks you mentioned, @r-ulak:

1. **Sending the Workflow Back to Previous Steps**: This can be achieved by designing your workflow in such a way that it can loop back to previous activities based on certain conditions or triggers.
2. **Reassigning the Step to Another Owner**: You might consider implementing a custom activity that allows reassignment based on specific criteria, such as the current assignee being unavailable.
3. **Hierarchical Approval**: As you rightly noted, this can be structured using a series of approval activities, each representing a level in the hierarchy.

@Nokecy, regarding your query about Elsa v3, the flexibility of Elsa means that with custom development, you should be able to achieve the desired functionalities.

I hope this helps.
